streams/src/main/java/org/apache/kafka/streams/kstream/SessionWindows.java

```diff
@@ -84,17 +86,14 @@ private SessionWindows(final long gapMs, final long graceMs) {
      * Create a new window specification with the specified inactivity gap.
      *
      * @param inactivityGap the gap of inactivity between sessions
-     * @return a new window specification with default maintain duration of 1 day
+     * @return a new window specification with default without any grace period
```

Comment:
I might have forgotten to specify this in the KIP, but we should make sure not to change the behavior of these old deprecated constructors. Maybe we can phrase this as something like:
```suggestion
     * @return a new window specification without specifying a grace period (uses the old default grace period of 24hr)
```

Reply:
> I might have forgotten to specify this in the KIP, but we should make sure not to change the behavior of these old deprecated constructors. Maybe we can phrase this as something like:

Hi @ableegoldman I may have misunderstood the KIP but I thought that we were changing org.apache.kafka.streams.kstream.Windows.DEFAULT_GRACE_PERIOD_MS from 24 hours (in milliseconds) to 0 milliseconds. That was what I did. Based on your feedback, it seems I misread the KIP and have the wrong implementation. 

So, should we still leave the org.apache.kafka.streams.kstream.Windows.DEFAULT_GRACE_PERIOD_MS constant as (24 * 60 * 60 * 1000L) milliseconds and only set the grace period of zero milliseconds in the new static constructors?

Reply:
Sorry, it's kind of a subtle point -- we would like to essentially get rid of the default, which means the grace period is always specified exactly by the user. In the `withNoGracePeriod` APIs, the user has just specified the grace period as 0 -- it's not a default 🙂 

On top of that, we actually can't change the current behavior of the existing APIs which we are deprecating here. That means those still need to set the grace period as the old default of 24hr. Once the deprecation period has passed, we can remove the DEFAULT_GRACE_PERIOD config when we remove the deprecated APIs 

Does that make sense? It might help to rename the DEFAULT_GRACE_PERIOD config to make this more clear, eg to something like DEPRECATED_GRACE_PERIOD or OLD_DEFAULT_GRACE_PERIOD -- WDYT?

Reply:
@ableegoldman  I will change it now to DEPRECATED_OLD_24_HR_GRACE_PERIOD and for the zero grace period, I will use NO_GRACE_PERIOD as the constant